setup.py

```diff
@@ -32,29 +35,23 @@
 with open("README.md") as readme_file:
     long_description = readme_file.read()
 
-strict_requirements = {
-    "numba": "numba>=0.4.8",
-    "matplotlib": "matplotlib==3.2.2",
-}
-with open("requirements.txt") as requirements_file:
-    requirements = []
-    # for line in requirements_file:
-    #     # TODO, this should be a proper regex parsing
-    #     requirement, version = line.split("==")
-    #     if requirement not in strict_requirements:
-    #         requirements.append(requirement)
-    #     else:
-    #         requirements.append(strict_requirements[requirement])
-    for line in requirements_file:
-        requirements.append(line)
 
-def is_tool(name):
-    """Check whether `name` is on PATH and marked as executable."""
-    from shutil import which
-    return which(name) is not None
-if os.name == 'posix':
-    if not is_tool('mono'): #Do not try to install pythonnet if mono is not installed.
-        requirements = [_ for _ in requirements if not _.startswith('pythonnet')]
+extra_requirements = {}
+for extra, requirement_file_name in package2install.__requirements__.items():
+    with open(requirement_file_name) as requirements_file:
+        if extra != "":
+            extra_stable = f"{extra}-stable"
+        else:
+            extra_stable = "stable"
```

Comment:
This is the point I have been hinting at before: even for the base file: `requirements.txt`, there is a stable and a normal version.